### ISSUE TITLE
Test webhook payload format incorrect

### ISSUE THREAD

**Issue by snoek8** [mautic/mautic#4151]
| Bug report | 

## Description:
Trying to set up webhooks with Zapier. The webhook test data sends in the correct format but the actual webhook data is in incorrect format.

### Steps to reproduce:
1. Set up a webhook catch on zapier
2. Run the test post from Mautic. You will see zapier picks up the fields correctly
3. Rerun the test in zapier but this time use an actual webhook post. You will see the data is sent like this:
mautic.lead_post_save_new:	contact: {u'modifiedByUser': None, u'dateAdded': u'2017-05-24T22:46:59-05:00',....etc.etc.}
timestamp: 2017-05-24T22:47:00-05:00

**Comment by escopecz** [comment]
Zapier is not supported by the core team. See https://github.com/mautic/mautic/issues/2838

**Comment by snoek8** [comment]
The issue is not about zapier integration. The problem is that the webhook
posts from Mautic are incorrect. The testpost from Mautic works correct but
actual post does not. That will not just affect zapier webhook catch but
others too.

**Comment by escopecz** [comment]
I tested it. If I test the payload I get this format:
```
{"mautic.lead_post_delete":{"lead":{"id":null,"points":0,"color":""...
```
The live webhook has this format:
```
{"mautic.lead_post_save_update":[{"lead":{"isPublished":true,"dateAdded":...
```
The difference is that the lead object is surrounded in one more array. This change was added to unify the response with other webhooks. For example post delete and on point change have this array because there is more data than just the lead object. So the test payload object has the wrong format. I don't know Zapier, but is it that hard to parse the lead object from that array?

I'll mark this as bug as the test payload doesn't have the same format as the live payloads.

**Comment by snoek8** [comment]
@escopecz I get the expected behavior when posting the test data but get invalid json when posting the live payload so my assumption is that the json for the live post is invalid and not the test data. Can you please review?

**Comment by snoek8** [comment]
@escopecz I just posted the live data to https://requestb.in and the resulting json is invalid. Looks like it is not correctly applying closing brackets. So maybe this bug is actually 2 issues. 
1. The test and live payload structures do not match
2. The live payload is invalid json

**Comment by snoek8** [comment]
Sure. Here you go. https://requestb.in/1dn6gp51?inspect

**Comment by escopecz** [comment]
I wasn't able to replicate the issue with incomplete JSON, but I updated the testing payloads to the new format at #4200. I'm also working on a Mautic app for Zapier at https://github.com/mautic/mautic-zapier. It should be finished and available in their app library within 2-3 weeks.

**Comment by snoek8** [comment]
Could the issue be caused by having custom fields maybe? I am constantly able to replicate the issue so it is strange you cannot replicate it so maybe the difference is custom fields?

**Comment by escopecz** [comment]
May be. How many custom fields do you have?

It can be also a limit on the database column. Do you use queued webhooks or immediate?

Or it can be limit of the POST request which is also configurable in your server config.

**Comment by snoek8** [comment]
Thank you. Turned out to be the max_input_vars value in php.ini. The default is set to 1000. I incremented the value and now the full json gets posted. Now all that is left is to make sure that the post format matches the test but I think you have that change implemented correct?

**Comment by escopecz** [comment]
Actually, the test now matches the real data webhook. As I wrote earlier, it's how it suppose to be. The other webhook triggers like page hit and form submit have the same format. The Zapier app will work with that format too.
